pkg/build/builder/cmd/scmauth/scmauth.go

```diff
@@ -11,5 +13,5 @@ type SCMAuth interface {
 	Handles(name string) bool
 
 	// Setup lays down the required files for this authentication method to work
```

Comment:
document what the returned URL is... if the auth methods mutate the URL now, their order becomes important... how is that determined? are there test cases around it?